### PR TITLE
Replace typography components with Text

### LINKED ISSUE #6588: [Typography foundations] Replace usage of deprecated type components
When the existing type components are deprecated, replace places they're used in polaris-react with the new Text component.

### LINKED ISSUE #7047: [Typography foundations] [Migration] Toasts
**[Figma](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=29653%3A21166)**

Update `toast` message text to use `variant="bodyMd" weight="medium"`

| Before | After |
| --- | --- |
| `DisplaySmall`Screen Shot 2022-08-30 at 8 55 51 AM | `BodyMd–medium`image |

### LINKED ISSUE #7048: [Typography foundations] [Migration] Form elements
Figma links:
- **[Text field](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=376%3A12)**
- [Select](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=364%3A2186)
- **[Filters](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=6307%3A12)**
- **[Top bar](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=434%3A2197)**

I assume any component with an input is probably consuming `TextField` or `Select`. I'm thinking about `Filters`, any search fields, `Combobox`, etc. but correct me if that's not right.

Update any input fields like: `TextField`, `Select`:
- Desktop: variant="bodyMd"
- Mobile: variant="bodyLg" for input text, labels, inline errors, and help text remain variant="bodyMd"

| Before | After |
| --- | --- |
| image | image |
|image|image|

### LINKED ISSUE #7192: [Typography foundations] [Migration] Action list
**[Figma](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=254%3A440)**

With updates:
image

- Update the top and bottom padding of Action list from 8px to 4px
- Move the divider to above section title when there are multiple sections

| Before | After |
| --- | --- |
| image | image |

### LINKED ISSUE #7197: [Typography foundations] [Migration] Description list
**[Figma](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=53638%3A30406)**

Update titles to HeadingSm:
| Before | After |
| --- | --- |
| image | image |

### LINKED ISSUE #7199: [Typography foundations] [Migration] Empty state

### LINKED ISSUE #7205: [Typography foundations] [Migration] Navigation

### LINKED ISSUE #7206: [Typography foundations] [Migration] Option list
[**Figma**](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=441%3A3720)
- Add more space above section titles when there are multiple sections. Update top padding above section title to 16px instead of 8px
- Update section title color to text-default instead of subdued
- add divider above section title (let me know if this is out of scope, we can leave off the divider if so - I basically did the same thing as action list here)

| Before | After |
| --- | --- |
| image | image |

### LINKED ISSUE #7583: [Typography foundations] [Migration] Badge
[Figma file](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=367%3A343)
Both medium and small variants will now use `body-sm`

image

### LINKED ISSUE #7584: [Typography foundations] [Migration] Card
[Figma](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=10%3A25)

`Card` section headings should use `heading-sm` (14px/20px)

The original intent was to use `heading-xs` but with current spacing and alignment with links that are 14px, `heading-sm` feels better.

image

### LINKED ISSUE #7598: [Typography foundations] [Migration] Page
**[Figma](https://www.figma.com/file/siMyneYEuULTaNSuRFZAcx/Type-migrations?node-id=345%3A1992)**

This might be taken care of manually but adding just in case - page title should stay the same across screen sizes. Right now it gets bigger on smaller screens.

image

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Resolves #6588, resolves #7192, resolves #7584, resolves #7047, resolves #7197, resolves #7048, resolves #7199, resolves #7206, resolves #7205, resolves #7598, resolves #7583.

### WHAT is this pull request doing?

Replaces the old typography components with the new `Text` component. These changes where created using the `replace-text-component` migration:

```sh
polaris-migrator replace-text-component "./polaris-react/src/**/*.tsx" --relative
```

Additionally, manual updates for the following text mixins have been updated to either use the `Text` component or the token inside the classes directly:

- **text-style-body**
- **text-style-heading**
- **text-style-subheading**
- **text-style-caption**
- **text-style-button**
- **text-style-button-large**
- **text-emphasis-subdued**
- **text-emphasis-strong**
- **nav-item-text-attributes**

### CODE REVIEW

File: polaris-react/src/components/Badge/tests/Badge.test.tsx
Comment (laurkim) [inline polaris-react/src/components/Badge/tests/Badge.test.tsx]:
```suggestion
    expect(badge).toContainReactComponent(Text, {visuallyHidden: true});
```
lint fix

**Comment by laurkim** [comment]
This is blocked until we add responsive styling to the `Text` component to accommodate the responsive styling that `DisplayText` has. I'll link the PR for that functionality once I have it up.

**Comment by sarahill** [comment]
@samrose3 there are a few components that need to be manually updated because they don't follow the mappings 1:1. There are details in individual issues for each component. Links to those issues can be found in #6794. I'll list them here too:
- #7192 
- #7197
-  #7198 
-  #7199 
- #7048
- #7205 
- #7206 
- #7047
- #7046 

There's details on why and what they should look like. There are some cases where we needed to make adjustments for smaller screens others where the styles just needed to be adjusted to visually work with the updated styles. Happy to chat more about this. Let me know how I can help. Thanks!

**Comment by laurkim** [comment]
> This is blocked until we add responsive styling to the `Text` component to accommodate the responsive styling that `DisplayText` has. I'll link the PR for that functionality once I have it up.

[PR](https://github.com/Shopify/polaris/pull/7538) to update responsive styling

**Comment by laurkim** [comment]
@samrose3 I rebased off latest `main` to pick up recent changes to `Text` that added responsive styling.

**Comment by aveline** [comment]
There are 86 changes in Chromatic, appears to be changes in font sizes. Are these expected? Have they all been reviewed? If so, they should also be accepted.

Personally some of them don't look right to me. In some cases if the larger font size is expected, then I think spacing needs to be adjusted.

cc @sarahill @nayeobkim

**Comment by sarahill** [comment]
Just chatted with @laurkim and she's going to add responsive styling for `HeadingXl` and `HeadingLg`. I think this should take care of some of the regressions. There will still be some because of the weight changes which are expected.

I think Lo mentioned the mapping will also need to be updated because it was a little off. 

@aveline @laurkim  Does it make sense to wait to go through the Chromatic changes until the responsive PR is merged and the mappings are updated?

Also, we decided to hold off on those manual updates until we QA the current state with the mappings updated then we can make a call on the priority of those manual updates.

**Comment by aveline** [comment]
https://github.com/Shopify/polaris/pull/7530#issuecomment-1293434805 Looks like this should now include the responsive styles, but yes wait until the mappings are updated before going through Chromatic.

**Comment by sam-b-rose** [comment]
I've updated the Display Text variant mapping and incorporated the responsive style changes. Everything should be g2g and ready for a code and visual review!

**Review by sarahill — DISMISSED** [review]
One thing I'm noticing as I'm reviewing Chromatic is that text using `body-md` seems to be going up to `15px` on `breakpoint-xs` and `breakpoint-sm`. `15px` isn't a value in our type scale.

The intent is that the default style, `body-md`, would stay the same across screen sizes by default. The exceptions are:
- `Heading4xl`, `Heading3xl`, `Heading2xl`, `HeadingXl`, `HeadingLg`
- and input fields should use size `16px` with line-height `20px` on `breakpoint-xs` and `sm`

☝️ These are looking as expected. It's just `body-md` that's using `15px`.

Let me know if you have questions. Happy to chat more and sorry for all the back and forth.

@samrose3 @alex-page is this in scope or is the intent to hardcode values, for now, to keep visual changes minimal? I'm unclear on the approach and it's making it hard to QA. I could be misremembering and if you can help me with getting some clarity I would appreciate it :)

**Comment by laurkim** [comment]
> One thing I'm noticing as I'm reviewing Chromatic is that text using `body-md` seems to be going up to `15px` on `breakpoint-xs` and `breakpoint-sm`. `15px` isn't a value in our type scale.

@sarahill this is probably because there are classNames wrapping the text component with the `text-style-body` mixin. I can take another pass to remove any I find.

**Comment by laurkim** [comment]
@sarahill I just pushed up a commit to fix components that were referencing `text-style-body` mixin. The components affected were: Breadcrumbs (which is interesting because they're visually hidden anyways), Choice and ChoiceList, EmptyState, Label/Labelled (used in Select and TextField with help text storybook examples), and Tabs.

File: polaris-react/src/components/Breadcrumbs/Breadcrumbs.scss
Comment (aveline) [inline polaris-react/src/components/Breadcrumbs/Breadcrumbs.scss]:
```suggestion
```

Doesn't look like this is needed anymore

File: polaris-react/src/components/DropZone/components/FileUpload/FileUpload.tsx
Comment (aveline) [inline polaris-react/src/components/DropZone/components/FileUpload/FileUpload.tsx]:
```suggestion
      <Text variant="bodyMd" color="subdued" as="span">
        {actionHint}
      </Text>
```

File: polaris-react/src/components/EmptyState/EmptyState.scss
Comment (aveline) [inline polaris-react/src/components/EmptyState/EmptyState.scss]: This should be removed and `color="subdued"` passed in to the `<Text>` component

File: polaris-react/src/components/EmptyState/EmptyState.tsx
Comment (aveline) [inline polaris-react/src/components/EmptyState/EmptyState.tsx]:
```suggestion
      <Text as="span" variant="bodyMd" color="subdued">
```

**Review by aveline — APPROVED** [review]
⚙️⚡️✨

**Comment by laurkim** [comment]
We're going to be updating mixins (i.e., `text-style-heading` and `text-emphasis-normal`) in this PR so I'm moving this back to draft until it's ready for review.

**Comment by laurkim** [comment]
In our next major release, we should remove support for `size` as a prop on `Badge`. The `size` prop is responsible for overriding the `font-size` but since `font-size` is now going to use the `bodySm` variant for both cases, it's no longer necessary. I didn't want this PR to be blocked from merging for the next major release, so once this PR is merged into `main`, I'll remove `size` support in the working v11 branch. More context [here](https://github.com/Shopify/polaris/issues/7583).

File: polaris-react/src/components/Banner/Banner.scss
Comment (sam-b-rose) [inline polaris-react/src/components/Banner/Banner.scss:263]: Could we use a `Text` component in the button's children JSX here instead of individual styles?

File: polaris-react/src/components/Button/Button.scss
Comment (sam-b-rose) [inline polaris-react/src/components/Button/Button.scss:52]: Curious why these `initial` declaration values are necessary?

**Comment by sarahill** [comment]
Started a spreadsheet to track regressions https://docs.google.com/spreadsheets/d/1JbtTb7PE-PtylsjccspWjOdqrYp15Ghdi1kau8ecNEc/edit?usp=sharing

File: .changeset/calm-wasps-cough.md
Comment (kyledurand) [inline .changeset/calm-wasps-cough.md]:
```suggestion
Replaced all typography components with the new `Text` component.
```

File: polaris-react/src/components/Badge/Badge.tsx
Comment (kyledurand) [inline polaris-react/src/components/Badge/Badge.tsx:56]: This is a bit weird because we're removing the styling for the small prop. Can we deprecate the size prop in this PR? 

Something like:

```tsx
  /** @deprecated Use the default size. */
  size?: Size;

File: polaris-react/src/components/Filters/Filters.tsx
Comment (kyledurand) [inline polaris-react/src/components/Filters/Filters.tsx]: Not a blocker, I'd consider a ternary inside of the color prop easier to read:

```suggestion
                  color={this.props.disabled || filter.disabled ? 'subdued' : undefined}
```

File: polaris-react/src/components/OptionList/OptionList.tsx
Comment (kyledurand) [inline polaris-react/src/components/OptionList/OptionList.tsx]: Same with the ternary here, especially since it would match the style on line 98. Again it's just my preference, this is easy enough to read

File: polaris-react/src/components/RangeSlider/components/DualThumb/DualThumb.scss
Comment (kyledurand) [inline polaris-react/src/components/RangeSlider/components/DualThumb/DualThumb.scss]: I wonder if we should use `> :first-child` like above. In case someone changes from span to another element

File: polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (kyledurand) [inline polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss]: Same here if you make the change above

File: polaris-react/src/components/Select/Select.scss
Comment (kyledurand) [inline polaris-react/src/components/Select/Select.scss:54]: This is beginning to get hairy. Can we keep any class names here?

File: polaris-react/src/components/Select/Select.scss
Comment (kyledurand) [inline polaris-react/src/components/Select/Select.scss]: Is this leftover from testing?

**Review by kyledurand — COMMENTED** [review]
This is huge, great job all. There are a few comments but nothing super pressing, except maybe `background-color: green` on the Select label. I'd like to fix up some of the nested element selectors where possible

**Review by sam-b-rose — COMMENTED** [review]
Awesome, work! Looks good to me.

There seem to be a few changes unrelated to the Text and text style mixins migration, however I'm comfortable with them as long as we have a thorough design review 👍

File: polaris-react/src/components/Select/Select.scss
Comment (laurkim) [reply polaris-react/src/components/Select/Select.scss]: Oops 🌝

File: polaris-react/src/components/Select/Select.scss
Comment (laurkim) [reply polaris-react/src/components/Select/Select.scss:54]: Jammed on this offline and decided to keep this as is and rely on Chromatic to catch any breaking layout changes.

File: polaris-react/src/components/Banner/Banner.scss
Comment (laurkim) [reply polaris-react/src/components/Banner/Banner.scss:263]: The primary action Button that uses this class uses the `unstyledButtonFrom` and passes it `action` prop that's being passed to `Banner` as content. I'm hesitant to update the util just so that it uses `Text` because the style of the text component will vary depending on use case.

File: polaris-react/src/components/Button/Button.scss
Comment (laurkim) [reply polaris-react/src/components/Button/Button.scss:52]: These came directly from the `text-style-button` [mixin](https://github.com/Shopify/polaris/blob/3438651591579ca741b1d42f8f58141065c1ec75/documentation/guides/legacy-polaris-v8-public-api.scss#L1927-L1938) that I updated to adhere to our new UI updates. I'm guessing they were added to the mixin to prevent inheriting styles from parent components.

**Review by sarahill — APPROVED** [review]
🎉 🚀 🥳

File: polaris-react/src/components/AccountConnection/AccountConnection.tsx
Comment (aveline) [inline polaris-react/src/components/AccountConnection/AccountConnection.tsx:64]: If we change the `as` to a `div` can the wrapper `div` be removed?

File: polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (aveline) [inline polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss:225]: Not related to this PR but when I see this it makes me wonder if we are missing a layout primitive?

**Review by aveline — COMMENTED** [review]
Nice work!

File: polaris-react/src/components/AccountConnection/AccountConnection.tsx
Comment (laurkim) [reply polaris-react/src/components/AccountConnection/AccountConnection.tsx:64]: We don't support `div` as an element type on `Text` 😞

File: polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss
Comment (laurkim) [reply polaris-react/src/components/RangeSlider/components/SingleThumb/SingleThumb.scss:225]: Great point! @kyledurand and I were discussing this and he started an [issue](https://github.com/Shopify/polaris/issues/7618) to track examples like this where we need to use selectors to get elements because we aren't able to apply styles with the classnames 👍

**Review by kyledurand — APPROVED** [review]
👏

**Pull request merged**
